benches/bandwidth.rs

```diff
@@ -0,0 +1,176 @@
+use std::convert::TryInto;
+
+use criterion::measurement::WallTime;
+use criterion::{black_box, criterion_group, criterion_main, BenchmarkGroup, Criterion};
+use rand::distributions::{Distribution, Standard};
+use rand::rngs::SmallRng;
+use rand::SeedableRng;
+use reed_solomon_erasure::galois_8::ReedSolomon;
+
+type Shards = Vec<Vec<u8>>;
+
+fn create_shards(block_size: usize, data: usize, parity: usize) -> Shards {
+    let mut small_rng = SmallRng::from_entropy();
+
+    let mut shards = Vec::new();
+
+    // Create data shards with random data
+    shards.resize_with(data, || {
+        Standard
+            .sample_iter(&mut small_rng)
+            .take(block_size)
+            .collect()
+    });
+
+    // Create empty parity shards
+    shards.resize_with(data + parity, || {
+        let mut vec = Vec::with_capacity(block_size);
+        vec.resize(block_size, 0);
+        vec
+    });
+
+    shards
+}
+
+fn rs_encode_benchmark(
+    group: &mut BenchmarkGroup<WallTime>,
+    block_size: usize,
+    data_shards: usize,
+    parity_shards: usize,
+) {
+    let size = block_size * data_shards;
+
+    group.throughput(criterion::Throughput::Bytes(size.try_into().unwrap()));
+
+    group.bench_function(format!("{}+{}", data_shards, parity_shards), |b| {
+        let mut shards = create_shards(block_size, data_shards, parity_shards);
+        let rs = ReedSolomon::new(data_shards, parity_shards).unwrap(); // 3 data shards, 2 parity shards
```

Comment:
Comment here and below is irrelevant since parameters are provided from the outside